src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -426,6 +435,37 @@ private void EnqueueDocumentChangedEvent(Solution oldSolution, Solution newSolut
                         var newProject = newSolution.GetRequiredProject(documentId.ProjectId);
 
                         await EnqueueChangedDocumentWorkItemAsync(oldProject.GetRequiredDocument(documentId), newProject.GetRequiredDocument(documentId)).ConfigureAwait(false);
+
+                        if (WorkspaceConfigurationOptions.ShouldConnectSourceGeneratedFilesToWorkspace(newSolution.Options))
```

Comment:
So this will mean that if a regular document changes we'll reanalyze source generated documents, but what about other triggering changes? For example, changing metadata references?

Reply:
Also are downstream projects handled via normal graph dependency, I'd hope?

Reply:
⏱️ Not sure, but it seems easy enough to test and will report if it works

Reply:
@sharwell We did test this?